### PR TITLE
QUARKUS-1223 Hibernate Search product support

### PR DESCRIPTION
**Pull request by jsmrcka**
https://issues.redhat.com/browse/QUARKUS-1223

Test plan for Hibernate Search feature provided by `hibernate-search-orm-elasticsearch` extension.
Guide: https://quarkus.io/guides/hibernate-search-orm-elasticsearch

### CODE REVIEW

**Review by rsvoboda — CHANGES_REQUESTED** [review]
I miss info about impact on execution env and execution time.

OpenShift and development experience topics are not covered well enough.

Are there any extensions combinations that should be checked, does this extension work well with both RESTEasy reactive and classic (upstream tests are using RESTEasy classing for the jaxrs endpoints)?

File: QUARKUS-1223.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - Multi-tenancy using separate schema
+  - https://github.com/quarkusio/quarkus/tree/main/integration-tests/hibernate-search-orm-elasticsearch (tests disabled by default)
+    - Dev Services
+    - Embedded indexes using `@IndexedEmbedded`
+  - https://github.com/quarkusio/quarkus/blob/main/integration-tests/hibernate-search-orm-opensearch (tests disabled by default)
+    - OpenSearch backend
+- Quickstarts: https://github.com/quarkusio/quarkus-quickstarts/tree/main/hibernate-search-orm-elasticsearch-quickstart
+    - Initial indexing using `MassIndexer`
+
+What is missing:
```
[the remark this answers is not in the document]
Comment (jsmrcka) [reply QUARKUS-1223.md:46]: After consultation with Yoann: 
- Multiple PUs are covered in https://github.com/quarkusio/quarkus/blob/main/extensions/hibernate-search-orm-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/elasticsearch/test/multiplepersistenceunits/MultiplePersistenceUnitsCdiTest.java (I missed that one). We can implement an IT in our TS, but I'm not sure if there would be any added value. 
- `@GenericField` and `Bridges` are not mentioned in the Quarkus guide, along with other features form HSearch. It's not practical to test everything from HSearch, so I propose to omit these features.

File: QUARKUS-1223.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+- Multiple persistence units
+- Use of `@GenericField`
+- Use of Bridges
+
+## Automated test development
+- Integrate Quickstart (https://github.com/quarkusio/quarkus-quickstarts/tree/main/hibernate-search-orm-elasticsearch-quickstart) into testing pipelines.
+- Adopt Quarkus integration tests into QE test suite or make sure the integration tests are part of upstream testing pipeline.
```
[the remark this answers is not in the document]
Comment (jsmrcka) [reply QUARKUS-1223.md:53]: My mistake, the upstream ITs are executed, so no need to duplicate them.

File: QUARKUS-1223.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- Quickstarts: https://github.com/quarkusio/quarkus-quickstarts/tree/main/hibernate-search-orm-elasticsearch-quickstart
+    - Initial indexing using `MassIndexer`
+
+What is missing:
+- Multiple persistence units
+- Use of `@GenericField`
+- Use of Bridges
+
+## Automated test development
+- Integrate Quickstart (https://github.com/quarkusio/quarkus-quickstarts/tree/main/hibernate-search-orm-elasticsearch-quickstart) into testing pipelines.
```
[the remark this answers is not in the document]
Comment (jsmrcka) [reply QUARKUS-1223.md]: Yes. I rephrased it.

File: QUARKUS-1223.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+- Adopt Quarkus integration tests into QE test suite or make sure the integration tests are part of upstream testing pipeline.
+- Implement test coverage in QE test suite for missing topics (multiple PUS, `@GenericField`, Bridges).
```
[the remark this answers is not in the document]
Comment (jsmrcka) [reply QUARKUS-1223.md]: See https://github.com/quarkus-qe/quarkus-test-plans/pull/57#discussion_r818449481.

**Comment by jsmrcka** [comment]
@rsvoboda PR updated. Changes are in a separate commit, which I will squash after we resolve all issues.

File: QUARKUS-1223.md [hunk 5]
```diff
@@ -28,30 +29,41 @@ Testing should include following topics:
...
 ## Automated test development
-- Integrate Quickstart (https://github.com/quarkusio/quarkus-quickstarts/tree/main/hibernate-search-orm-elasticsearch-quickstart) into testing pipelines.
-- Adopt Quarkus integration tests into QE test suite or make sure the integration tests are part of upstream testing pipeline.
-- Implement test coverage in QE test suite for missing topics (multiple PUS, `@GenericField`, Bridges).
+- Enable relevant Quickstart module in Quickstart acceptance job: https://github.com/quarkusio/quarkus-quickstarts/tree/main/hibernate-search-orm-elasticsearch-quickstart.
+- Implement test coverage in QE test suite:
+  - OpenShift scenario.
+  - Combination with extensions:
+    - RESTEasy Reactive counterparts of RESTEasy extensions used in upstream tests.
+    - Other supported databases: `jdbc-derby`, `jdbc-mariadb`, `jdbc-mssql`, `jdbc-mysql`, `jdbc-oracle` (module `sql-db/sql-app` might be a good target location).
```
Comment (rsvoboda) [inline QUARKUS-1223.md:62]: Please do not mix stuff with `sql-db/sql-app` and create new module. It's says Hibernate Search but it doesn't have too much in common with vanilla Hibernate ORM.

I would pick 2PUs: `jdbc-mariadb` and  `jdbc-oracle`, PostgreSQL is covered in upstream.

File: QUARKUS-1223.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+What is missing:
+- Use of RESTEasy Reactive.
+- Use of other supported databases.
+
+## Automated test development
+- Enable relevant Quickstart module in Quickstart acceptance job: https://github.com/quarkusio/quarkus-quickstarts/tree/main/hibernate-search-orm-elasticsearch-quickstart.
+- Implement test coverage in QE test suite:
+  - OpenShift scenario.
+  - Combination with extensions:
```
Comment (rsvoboda) [inline QUARKUS-1223.md:60]: Ideally I would like to have combination of these use-cases: Multiple persistence units ++ Outbox polling ++ Multi-tenancy using separate schema
Ideally all of the combined, but we could end up with 2 of them if there are challenges with it.

**Review by rsvoboda — CHANGES_REQUESTED** [review]

**Comment by rsvoboda** [comment]
Please squash them and I will merge it

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**